Lib/urllib/parse.py

```diff
@@ -405,6 +402,16 @@ def _splitparams(url):
         i = url.find(';')
     return url[:i], url[i+1:]
 
+def _validate_port(port_str):
+    try:
+        port = int(port_str, 10)
+    except ValueError:
+        message = f'Port could not be cast to integer value as {port_str!r}'
+        raise ValueError(message) from None
+    if not (0 <= port <= 65535):
```

Comment:
Since this is TCP, 0 port should not be valid AFAIK.

Reply:
(but as it's a small backwards compatibility break, perhaps it should be a separate PR, I'm not sure)

Reply:
This has been discussed before and it was opted out to leave as it is.

https://github.com/python/cpython/pull/16780